HW_01.cpp

```diff
@@ -0,0 +1,133 @@
+#include <cmath>
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+
+uint8_t equalation_init(double *a, double *b, double *c);
+
+uint8_t equalation_solve(double a, double b, double c, double *x1, double *x2);
+
+void equalation_result_print(double a, double b, double c, uint8_t roots_num,
+                             double x1, double x2);
+
+int main() {
+
+  double a{};
+  double b{};
+  double c{};
+
+  if (!equalation_init(&a, &b, &c)) {
+    return EXIT_FAILURE;
+  }
+
+  double x1{};
+  double x2{};
+
+  uint8_t roots = equalation_solve(a, b, c, &x1, &x2);
+  equalation_result_print(a, b, c, roots, x1, x2);
+
+  return EXIT_SUCCESS;
+}
+
+/**
+ * @brief This function creates an equalation with specific A,B,C coefficient
+ *        Ax^2 + Bx + C = 0
+ *
+ * @param a        IO pointer to A coefficient
+ * @param b        IO pointer to B coefficient
+ * @param c        IO pointer to C coefficient
+ * @return uint8_t SUCCESS - 1, FAILURE - 0
+ */
+uint8_t equalation_init(double *a, double *b, double *c) {
+  if (!a || !b || !c) {
```

Comment:
рекомендую писать так
```suggestion
  if (a == nullptr|| b == nullptr || c == nullptr) {
    return 0;
  }
```
это добавит читабельности что вы юзаете именно указатели а не просто, например, интовые переменные

Reply:
Хорошо, исправлю